### PR TITLE
🐛(ci) fix npm development version

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

setuptools_scm generates a development version which can not be compatible with the semver syntax used by npm. For a development version, the version and prerelease tag should be seperate with a dash, setuptools_scm use a period as separation. We have to transform the version with one compatible with NPM.

## Proposal

- [x] fix npm development version

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -849,6 +849,11 @@ jobs:
             if [[ -z "${CIRCLE_TAG+x}" ]]; then
               python -m pip install --upgrade setuptools_scm
               NEXT_VERSION=$(python -m setuptools_scm)
+              # Match version like 1.2.3.dev123
+              if [[ $NEXT_VERSION =~ ([[:digit:]]+?.[[:digit:]]+?.[[:digit:]]+)?.([[:alnum:]]+) ]]; then
```
Comment (jbpenrath) [inline .circleci/config.yml:853]: You should escape the `.` char as currently you are matching any chars.

```suggestion
              if [[ $NEXT_VERSION =~ ([[:digit:]]+?\.[[:digit:]]+?\.[[:digit:]]+)?\.([[:alnum:]]+) ]]; then
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -849,6 +849,11 @@ jobs:
...
+                # transform it in 1.2.3-dev123
+                NEXT_VERSION="${BASH_REMATCH[1]}-${BASH_REMATCH[2]}"
```
Comment (jbpenrath) [inline .circleci/config.yml:855]: Nice, one thing learned today !

**Review by kernicPanel — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (lunika) [reply .circleci/config.yml]: Oh you are right. I though that the question mark can be used for this in bash. It is not the case

**Comment by kernicPanel** [comment]
Just a question, do we always get `dev` word in the version ? If so, maybe it would be easier to use sed.
```
NEXT_VERSION=$(python -m setuptools_scm | sed 's/.dev/-dev/')
```

**Comment by lunika** [comment]
> Just a question, do we always get `dev` word in the version ? If so, maybe it would be easier to use sed.
> 
> ```
> NEXT_VERSION=$(python -m setuptools_scm | sed 's/.dev/-dev/')
> ```

No it can be other world like `alpha`, `beta`, etc

**Comment by kernicPanel** [comment]
> > Just a question, do we always get `dev` word in the version ? If so, maybe it would be easier to use sed.
> > ```
> > NEXT_VERSION=$(python -m setuptools_scm | sed 's/.dev/-dev/')
> > ```
> 
> No it can be other world like `alpha`, `beta`, etc

Too bad

**Pull request merged**